Source/WebKit/Platform/cocoa/WebPrivacyHelpers.mm

```diff
@@ -347,8 +358,12 @@ void requestLinkDecorationFilteringData(LinkFilteringRulesCallback&& callback)
             RELEASE_LOG_ERROR(ResourceLoadStatistics, "Failed to request storage access quirks from WebPrivacy.");
         else {
             auto quirks = [data quirks];
-            for (WPStorageAccessPromptQuirk *quirk : quirks)
-                result.append(WebCore::OrganizationStorageAccessPromptQuirk { quirk.name, domainPairingsDictToMap(quirk.domainPairings) });
+            for (WPStorageAccessPromptQuirk *quirk : quirks) {
+                if ([quirk respondsToSelector:@selector(quirkDomains)] && [quirk respondsToSelector:@selector(triggerPages)])
```

Comment:
Nit - I think you can move the `-respondsToSelector:` call outside the loop (and use `+instancesRespondToSelector:` instead), to avoid the repeated runtime selector check.